client/app/lib/components/RadioButton.tsx

```diff
@@ -0,0 +1,38 @@
+import { FormControlLabel, Radio, Typography } from '@mui/material';
+
+interface RadioButtonProps {
+  value: string;
+  label: string;
+  className?: string;
+  description?: string;
+  disabled?: boolean;
+}
+
+/**
+ * To be used within `<RadioGroup>` wrappers in forms.
+ */
+const RadioButton = (props: RadioButtonProps): JSX.Element => {
+  return (
+    <div className="w-full">
+      <FormControlLabel
+        control={<Radio className="py-0 px-4" />}
+        value={props.value}
+        label={props.label}
+        className={props.className}
+        disabled={props.disabled}
+      />
+
+      {props.description && (
+        <Typography
+          variant="body2"
+          color={props.disabled ? 'text.disabled' : 'text.secondary'}
+          className="ml-[34px]"
+        >
+          {props.description}
+        </Typography>
+      )}
+    </div>
+  );
+};
+
+export default RadioButton;
```

Comment:
I did not put `RadioButton` in `form/fields` because it is not a form component, i.e., it does not take `field` or `fieldState` from react-hook-form. It is just like `Checkbox` (in #5154) and `TextField`; they are plain bare components.

Thinking about it, there is no such thing as 'radio button field', because a radio button cannot have a state of its own. If we were to put anything in `form/fields`, it should be a `RadioGroup` that contains many `RadioButton`s, because that is where the `onChange` and `value` (spread from `field`) come from.